packages/synapse-react-client/src/components/OrientationBanner/OrientationBanner.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import { useTheme } from '@mui/material'
+import FullWidthAlert, {
+  AlertButtonConfig,
+} from '../FullWidthAlert/FullWidthAlert'
+import Illustrations from '../../assets/illustrations'
+
+const OrientationBannerNameStrings = [
+  'Challenges',
+  'CohortBuilder',
+  'DataAccessManagement',
+  'Datasets',
+  'Discussions',
+  'Docker',
+  'Favorites',
+  'Files',
+  'Following',
+  'Projects',
+  'Tables',
+  'Teams',
+  'TrashCan',
+  'Wikis',
+] as const
+export type OrientationBannerName =
+  (typeof OrientationBannerNameStrings)[number]
+
+export function getOrientationBannerKey(name: OrientationBannerName) {
+  return 'orientation_banner_' + name.toLowerCase() + '_dismissed'
+}
+
+export const ORIENTATION_BANNER_KEYS = OrientationBannerNameStrings.map(el =>
+  getOrientationBannerKey(el),
+)
+
+export interface OrientationBannerProps {
+  name: OrientationBannerName
+  title: string
+  text: React.ReactNode
+  primaryButtonConfig?: AlertButtonConfig
+  secondaryButtonConfig?: AlertButtonConfig
+}
+
+function OrientationBanner(props: OrientationBannerProps) {
+  const { name, title, text, primaryButtonConfig, secondaryButtonConfig } =
+    props
+
+  const storageBannerId = getOrientationBannerKey(name)
+  const [showBanner, setShowBanner] = React.useState(
+    localStorage.getItem(storageBannerId) === null,
+  )
```

Comment:
Show the banner if localStorage doesn't contain this banner's key, i.e. the banner hasn't previously been dismissed